[exporter/debug] Add file output options for customLogger (#13066)
open-telemetry/opentelemetry-collector: exporter/debugexporter/README.md

  Refer to [Zap docs](https://godoc.org/go.uber.org/zap/zapcore#NewSampler) for more details
  on how sampling parameters impact number of messages.
- `use_internal_logger` (default = `true`): uses the collector's internal logger for output. See [below](#using-the-collectors-internal-logger) for description.

> - `output_paths`: (optional, only used if `use_internal_logger: false`) list of file paths to write logging output to. Defaults to `["stdout"]`. Special strings are "stdout" and "stderr".

Reviewer: It's not "logging" output, because it is not ["logging" exporter](https://words.boten.ca/byebye-logging-exporter/) :)
```suggestion
- `output_paths`: (optional, only used if `use_internal_logger: false`) list of file paths to write output to. Defaults to `["stdout"]`. Special strings are "stdout" and "stderr".
```

> ## Example configurations:

Reviewer: Headers shouldn't end with a colon
```suggestion
## Example configurations
```

> ### Using collector's internal telemetry pipeline and logger

Reviewer: I'm worried that this may be too much information here. The "Example configurations" section should make it easy for users to start using a component they have no experience with. The header "Using collector's internal telemetry pipeline and logger" is quite a mouthful and in my opinion may be hard to digest by a novice user.

I propose to drop this header and also remove the `service` configuration and the `use_internal_logger: true` from the first example config, let's keep the first example simple.

The second example that uses `use_internal_logger: false` and `output_paths` looks good to me as a more advanced example for users looking for more.

  Refer to [Zap docs](https://godoc.org/go.uber.org/zap/zapcore#NewSampler) for more details
  on how sampling parameters impact number of messages.
- `use_internal_logger` (default = `true`): uses the collector's internal logger for output. See [below](#using-the-collectors-internal-logger) for description.

> - `output_paths`: (optional, only used if `use_internal_logger: false`) list of file paths or URLs to write logging output to. Defaults to `["stdout"]`.

Reviewer: Let's not mention "URLs", it may be confusing to users. I propose to officially support only bare file paths, regardless of what is [internally supported by Zap](https://pkg.go.dev/go.uber.org/zap@v1.27.0#Open).

Let's also mention the only special strings: "stdout" and "stderr".
```suggestion
- `output_paths`: (optional, only used if `use_internal_logger: false`) list of file paths to write logging output to. Defaults to `["stdout"]`. Special strings are "stdout" and "stderr".
```

    verbosity: detailed
    sampling_initial: 5
    sampling_thereafter: 200

>     use_internal_logger: false

Author: done

>     output_paths:

Reviewer: I'm not sure if it's needed to be able to set multiple output paths or just one path would suffice, but I suppose we can leave it as is.

I suppose one practical scenario is to send output to both `stdout` and a file on disk.

Reviewer: Works for me :+1:

      - /tmp/debugexporter.log

>     error_output_paths:

Reviewer: This option should not exist, see another comment.
